azurelinuxagent/ga/exthandlers.py

```diff
@@ -830,7 +904,16 @@ def uninstall(self):
             self.launch_command(uninstall_cmd)
         except ExtensionError as e:
             self.report_event(message=ustr(e), is_success=False)
-    
+
+    def rm_ext_pkg_zip(self):
+        try:
+            fn = self.try_get_pkg_file()
+            if fn is not None and os.path.exists(fn):
+                self.logger.info("Remove extension package: {0}".format(fn))
+                os.remove(fn)
+        except IOError as e:
```

Comment:
os.remove() is documented as raising OSError if anything goes wrong, in Python 2.7 and in 3.5. Later versions of Python alias IOError to OSError. as well. Upshot: this should be OSError.

Reply:
I should change the code I copied from then, done.